lib/features/profiles/screens/own_profile_screen.dart

```diff
@@ -11,29 +10,51 @@ import 'package:gruene_app/features/profiles/widgets/profile_header.dart';
 import 'package:gruene_app/i18n/translations.g.dart';
 import 'package:gruene_app/swagger_generated_code/gruene_api.swagger.dart';
 
-class OwnProfileScreen extends StatelessWidget {
+class OwnProfileScreen extends StatefulWidget {
   const OwnProfileScreen({super.key});
 
+  @override
+  State<OwnProfileScreen> createState() => _OwnProfileScreenState();
+}
+
+class _OwnProfileScreenState extends State<OwnProfileScreen> {
+  late Future<Profile?> _profileFuture;
+  Profile? _currentProfile;
+
+  @override
+  void initState() {
+    super.initState();
+    _profileFuture = fetchOwnProfile();
+  }
+
+  void _updateProfile(Profile updatedProfile) {
+    setState(() {
+      _currentProfile = updatedProfile;
+    });
+  }
+
   @override
   Widget build(BuildContext context) {
     return FutureLoadingScreen(
-      load: fetchOwnProfile,
-      buildChild: (Profile? data) {
-        if (data == null) {
-          return ErrorScreen(error: t.profiles.noResult, retry: fetchOwnProfile);
-        }
-
+      load: () async {
+        final profile = await _profileFuture;
+        _currentProfile ??= profile;
+        return _currentProfile!;
+      },
+      buildChild: (Profile data) {
```

Comment:
:wrench: Am I understanding this code correctly that you are duplicating the state here to be able to update the users profile? If yes, I think it would be better to handle this in the FutureLoadingScreen and expose an `update` function to the builder:
```suggestion
      load: fetchOwnProfile,
      buildChild: (Profile data, update) {
```
I think this might also be useful/necessary for future widgets and features.
L7 of FutureLoadingScreen (not sure if there is a way to avoid having to declare the variable in every usage):
``` dart
  final Widget Function(T data, void Function(T newData) update) buildChild;
```

Reply:
This is a very good idea, I implemented it this way.